packages/api/lambdas/sf-event-sqs-to-db-records/index.js

```diff
@@ -145,7 +165,18 @@ const handler = async (event) => {
       return await writeRecords({ ...event, cumulusMessage, knex });
     } catch (error) {
       log.error(`Writing message failed: ${JSON.stringify(message)}`, error);
-      return sendSQSMessage(process.env.DeadLetterQueue, message);
+      if (!process.env.DeadLetterQueue) {
+        log.error('DeadLetterQueue not configured');
+        return undefined;
+      }
+      return sendSQSMessage(
+        process.env.DeadLetterQueue,
+        {
+          ...message,
+          error: error.toString(),
+        },
+        log
```

Comment:
That's a good question - functionally it's changing sender metadata from '@cumulus/aws-client/SQS' to '@cumulus/api/lambdas/sf-event-sqs-to-db-records'.   

There's probably a reasonable question re: why that change was made and/or if we want the logger to identify the lambda or the portion of code.     I have opinions, but it feels like something that should be a deliberate change/pattern.

Reply:
Actually, looking at this the issue is the typing in sendSQSMessage is incorrect, it doesn't have the log override as an optional param.  I'd vote for fixing that over this change, however <scope> is a thing.

Reply:
it wasn't.
I thoguht originally that because the logger wasn't an optional it was required to pass in the log at the scope of whatever's calling it but... on looking further it ismeant to be optional